<?xml version="1.0"?>
<launch>
  <arg name="side" default="r"/> <!-- "b" = blue side, "r" = red side  -->
  <arg name="robot_namespace" default=""/>
  <arg name="rviz_file" default="burger_navigation.rviz"/>
    <!-- You can change control program  -->

  <!-- sample program node -->

    <include file="$(find burger_navigation)/launch/burger_navigation.launch">
      <arg name="robot_namespace" default="$(arg robot_namespace)"/>
      <arg name="rviz_file" value="$(arg rviz_file)"/>

      <!--  カクついたらここをコメント解除　-->
      <!-- <arg name="cmd_vel_topic" value="/move_base/cmd_vel"/> -->
    </include>

      <!--  カクついたらここをコメント解除　-->
    <!-- <node pkg="burger_war" type="cmd_smoother.py" name="cmd_smoother" output="screen"/> -->



  <!-- <node pkg="burger_war" type="seigoRun1.py" name="seigoRun" output="screen">
      <param name="robot_name" value="$(arg side)" />
    </node> -->

  <node pkg="burger_war_dev" type="seigoRun2.py" name="seigoRun" output="screen">
    <param name="side" value="$(arg side)" />
    <param name="robot_namespace" value="$(arg robot_namespace)" />

    <!-- 何秒前までの敵検出結果を有効とするか -->
    <param name="detect_enemy_time_tolerance" value="0.5" />

    <!-- 何メートル以内に敵が来たら攻撃行動に遷移するか -->
    <param name="snipe_distance_th" value="0.8" />

    <!-- この範囲内に障害物があるときは回復行動に移る -->
    <param name="distance_to_wall_th" value="0.15" />

    <!-- 攻撃するとき敵まで何メートル近づくか -->
    <param name="approach_distance_th" value="0.53" />

    <!-- 敵と方位角の角度差分がこのしきい値を下回ってから、接近 -->
    <param name="attack_angle_th" value="0.785" />

    <!-- escape parameter -->
    <!-- enable escape flag -->
    <param name="enable_escape_approach" value="True" />
     <!-- minimum distance to enemy(m) -->
    <param name="escape_approach_distance_th_min" value="0.231" />
    <!-- maxmum distance to enemy(m) -->
    <param name="escape_approach_distance_th_max" value="0.851" />
    <!-- interval time to min-max distance (sec) -->
    <param name="escape_approach_time_interval" value="6" />
  </node>

  <include file="$(find burger_war_dev)/launch/obstacle_detector.launch">
    <arg name="robot_namespace" default="$(arg robot_namespace)"/>
  </include>
  <node pkg="burger_war_dev" type="enemy_detector.py" name="enemy_detector_node" output="screen">
    <param name="robot_namespace" value="$(arg robot_namespace)"/>
  </node>
  <!-- END sample program node -->

  <!-- End of your space  -->
</launch>
